### PR TITLE
chore: reorganize API documentation

### PR DESCRIPTION
**Pull request by trujillo-adam**
This PR contains changes from the last successful commit in PR 3659. The changes present separate pages for the resources in each language in the API reference. This is a temporary fix for the documentation until @DanielMSchmidt  finishes updating the API reference documentation generator.

### CODE REVIEW

File: tools/documentation-generation/generate-documentation.js [hunk 1]
```diff
@@ -4,97 +4,203 @@
...
-      const fullTag = tag.substring(0, tag.indexOf(">"));
+import { Documentation, Language } from "jsii-docgen";
 
-      return tag.replace(fullTag, ` ${fullTag} `);
-    });
+(async function () {
+  const remarkParse = (await import("remark-parse")).default;
+  const remarkStringify = (await import("remark-stringify")).default;
+  const visit = (await import("unist-util-visit")).visit;
+  const unified = (await import("unified")).unified;
```
Comment (ansgarm) [inline tools/documentation-generation/generate-documentation.js:16]: Do we need to add these packages as dependencies in `tools/documentation-generation/package.json`?

File: website/data/cdktf-nav-data.json [hunk 2]
```diff
@@ -173,27 +173,132 @@
     "routes": [
       {
         "title": "Overview",
-        "path": "api-reference/overview"
```
Comment (ansgarm) [inline website/data/cdktf-nav-data.json:176]: Can / Should we redirect this to just `api-reference` for old links to still work?

File: website/data/cdktf-nav-data.json [hunk 3]
```diff
@@ -173,27 +173,132 @@
...
+        "path": "api-reference"
       },
       {
         "title": "Typescript",
-        "path": "api-reference/typescript"
```
Comment (ansgarm) [inline website/data/cdktf-nav-data.json:180]: Same as for the overview page – can / should we add redirects to just `api-reference`?

**Review by ansgarm — COMMENTED** [review]
Looks good! Nice work 👍
Only a smell question about the new dependencies and one about the now 404-ing paths

File: tools/documentation-generation/generate-documentation.js (hunk 1 above)
Comment (DanielMSchmidt) [reply tools/documentation-generation/generate-documentation.js:16]: Yeah, I probably added them in the next commit or so on the original branch. I'll also add them here 👍

**Review by ansgarm — APPROVED** [review]

**Comment by DanielMSchmidt** [comment]
Merging this, @trujillo-adam could you do a quick follow up with the redirects @ansgarm mentioned (we have no idea how those work and figured maybe you do 🙈 )

**Pull request merged**